opennms-webapp-rest/src/test/java/org/opennms/web/rest/v2/MonitoringLocationRestServiceIT.java

```diff
@@ -164,4 +172,68 @@ public void testEventOnCreationAndDeletion() throws Exception {
         // delete the one with polling packages
         sendRequest(DELETE, "/monitoringLocations/location2", 204);
     }
+
+    @Test
+    @Transactional
+    public void testLocationLimits() throws Exception {
+        // There is one default location
+        for (int i = 0; i < LOCATION_COUNT - 1; i++) {
+            Location loc = new Location();
+            loc.locationName = "Location-" + String.format("%05d", i);
+            loc.monitoringArea = "LocationArea-" + String.format("%05d", i);
+
+            // post it to web service
+            String location = "<location location-name=\"" + loc.locationName + "\" monitoring-area=\"" + loc.monitoringArea + "\"/>";
+            sendPost("/monitoringLocations", location, 201, null);
+        }
+
+        // Map contains (limit, expected count)
+        Map<Integer,Integer> limits = new HashMap<Integer,Integer>(8);
+        limits.put(null,DEFAULT_LIMIT); // null limit, should return default
+        limits.put(DEFAULT_LIMIT - 1,DEFAULT_LIMIT - 1); // limit less than default limit
+        limits.put(DEFAULT_LIMIT,DEFAULT_LIMIT); // limit equals default
+        limits.put(DEFAULT_LIMIT + 1,DEFAULT_LIMIT + 1); // limit greater than default
+        limits.put(LOCATION_COUNT + 1,LOCATION_COUNT); // limit greater than count
+        limits.put(0,LOCATION_COUNT); // unlimited, should return all
+
+        for (Integer limit: limits.keySet()) {
+            Map<String, String> parameters = new HashMap<String, String>(1);
+            if (limit != null) {
+                parameters.put("limit", limit.toString());
+            }
+            String fetchedJson = sendRequest(GET, "/monitoringLocations", parameters, 200);
+            Integer count = (new ObjectMapper()).readValue(fetchedJson, Locations.class).count;
+            LOG.info("limit="+limit+" expected="+limits.get(limit)+" count="+count);
+            if (limit == null) {
+                assertEquals(DEFAULT_LIMIT, count);
+            } else {
+                assertEquals(limits.get(limit),count);
+            }
+        }
+    }
+
+    /**
+     * Used for mapper to load Json from rest service
+     */
+    private static class Location {
+        public int priority;
+        public ArrayList<Object> tags;
+        public Object geolocation;
+        public Object latitude;
+        public Object longitude;
+        @JsonProperty("location-name")
+        public String locationName;
+        @JsonProperty("monitoring-area")
+        public String monitoringArea;
+    }
+
+    /**
+     * Used for mapper to load Json from rest service
+     */
+    private static class Locations {
```

Comment:
There is `OnmsMonitoringLocationDefinitionList` for this. Or just `JaxbListWrapper<OnmsMonitoringLocation>`.

Reply:
This gives:

Unrecognized field "location" (class org.opennms.web.rest.v1.support.OnmsMonitoringLocationDefinitionList), not marked as ignorable (4 known properties: "offset", "objects", "count", "totalCount"])